operations/kaas/cluster-stack-operator.md

```diff
@@ -0,0 +1,692 @@
+
+### User Story 3: Effortless Access to the Latest Cluster Stack Versions via Github Integration
+
+I expect the cluster-stack-operator to be endowed with the ability to continuously poll Github releases to identify any new releases that might be available. As a user who values automation, when I activate the "auto-subscribe" functionality, I anticipate the operator to not just detect the latest release but also to execute all requisite steps to make this release fully operational and ready for my use.
+
+### User Story 4: Facilitated Use of Specific, Legacy Versions of Cluster Stacks
+
+As a user with diverse requirements, there might be instances where I might require a specific older version of a cluster stack. For such scenarios, I should be equipped to delineate specific release versions within the objects I deploy. This precision ensures that my desired versions remain accessible for my use, allowing for flexibility and backward compatibility.
+
+### User Story 5: Personalized Configuration of Provider-specific Parameters, such as Node Images
+
+As a discerning user, I wish to define provider-specific configurations to suit my unique needs. Within the framework provided by the API of the respective provider-specific operator, I should have the freedom to stipulate details like the exact node images I prefer. Such control is instrumental in optimizing resources, ensuring I can steer clear of redundant node images, and potentially avoid tasks like unnecessary uploads.
+
+### User Story 6: Local Testing and Building of New Cluster Stacks for Providers
+
+Recognizing the pivotal role of providers, it's essential that they possess the capability to develop and validate their cluster stacks locally. The cluster-stack-operator should be intelligent enough to discern the source of release artifacts. If these are furnished locally, the operator should abstain from downloading them from external sources like Github. Such a feature is paramount for providers—it ensures they can construct, test, and validate their cluster stacks in a local environment, guaranteeing robustness and operational excellence.
+
+## Background
+
+### Understanding the Three Core Components of a Cluster Stack
+
+In the realm of cluster management and orchestration, the cluster stack is designed to seamlessly integrate and operate various functionalities. There are three pivotal components of a cluster stack:
+
+
+There are three components of a cluster stack:
+
+1. **Cluster Addons:** Cluster addons are a set of essential extensions and tools, including but not limited to the Cluster Network Interface (CNI), Container Storage Interface (CSI), and Cloud Controller Manager (CCM). These addons play a pivotal role in enhancing the functionality and manageability of each workload cluster initiated by the user. It's paramount to ensure that these addons are applied to every single workload cluster upon initiation, thus enabling seamless communication and operational consistency.
```

Comment:
As a more fundamental question: Maybe it makes sense to take one step back an review if at least part of the requirements should be implemented with tools already introduced to SCS, like flux2 [1]? This would make this document split into two parts as well as for flux maintenance will be more focused to setting up manifests.

It seems to me that partly the intended controller might reinvent the wheel.

[1] https://github.com/SovereignCloudStack/k8s-cluster-api-provider/pull/123

Reply:
We take an operator approach and it is not clear how this approach would integrate with flux to me. 
The described ClusterAddons and their controller would make it possible to install the cluster addons without any manual work or configuration. 

Additionally, we have some values from Cluster API objects that we inject into the yaml config of the objects we apply. I wouldn't know how to do that with external software either. 

Can you maybe elaborate on your idea to integrate flux?